File: CompetitionCSh/CompetitionCSh/domain/entities/SystemUser.cs
using CompetitionCSh.domain.interfaces;

namespace CompetitionCSh.domain.entities;

public class SystemUser : IHasId<string>
{
    public SystemUser(string userName, string password, string firstName, string lastName)
    {
        UserName = userName;
        Password = password;
        FirstName = firstName;
        LastName = lastName;
    }

    public string UserName { get; set; }
    public string Password { get; set; }

    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Id { get; set; }

    public override string ToString()
    {
        return
            $"SystemUser id {Id} ,userName {UserName}, password {Password}, firstName {FirstName}, lastName {LastName}";
    }
}

File: Competition/CompetitionsModel/src/main/java/competition/SystemUser.java
package competition;

import java.io.Serializable;
import java.util.Objects;

public class SystemUser implements HasId<Long>, Serializable {

    private Long id;

    private String username;
    private String password;

    private String firstName;
    private String lastName;


    public SystemUser() {
    }

    public SystemUser(String username, String password, String firstName, String lastName) {
        this.username = username;
        this.password = password;
        this.firstName = firstName;
        this.lastName = lastName;
    }

    @Override
    public Long getId() {
        return id;
    }

    @Override
    public void setId(Long newId) {
        this.id = newId;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof SystemUser)) return false;
        SystemUser that = (SystemUser) o;
        return getUsername().equals(that.getUsername()) && getPassword().equals(that.getPassword());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getUsername(), getPassword());
    }

    @Override
    public String toString() {
        return "SystemUser{" +
                "id='" + id + '\'' +
                ", username='" + username + '\'' +
                ", password='" + password + '\'' +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                '}';
    }
}
